modules/core/src/main/java/org/apache/fluo/core/client/FluoAdminImpl.java

```diff
@@ -516,6 +517,20 @@ public boolean oracleExists() {
     return oracleExists(getAppCurator());
   }
 
+  public int numOracles() {
```

Comment:
@keith-turner looking at the `oracleExists` method I wonder if we should switch `!curator.getChildren().forPath(ZookeeperPath.ORACLE_SERVER).isEmpty()` with something like 
`(new LeaderLatch(curator, ZookeeperPath.ORACLE_SERVER)).getParticipants().size() > 0`. 
The current implementation is probably right but it does rely on the LeaderLatch implementation and uses the LeaderLatch's path which Curator advises against (https://cwiki.apache.org/confluence/display/CURATOR/TN7). They're probably more concerned about creating and deleting nodes so it's probably not a big deal, but may be slightly safer.

Reply:
I think that would be a good change.